src/Database/Type/JsonType.php

```diff
@@ -102,4 +107,18 @@ public function marshal($value)
     {
         return $value;
     }
+
+    /**
+     * Set json_encode options.
+     *
+     * @param  int $jsonOptions Options. Use JSON_* flags. Set `0` to reset.
+     * @return self
```

Comment:
```suggestion
     * @return $this
```

Reply:
About $this. PhpStorm recommend `@return self` and `: self` as return type.

```php
     * @return self
     * @see https://www.php.net/manual/en/function.json-encode.php
     */
    public function setEncodingOptions(int $jsonOptions): self
```
With this, there is no warning on method.

Reply:
This is for PHP 8+, for lower PHP versions we still don't do this for several reasons, including chaining in the IDE
The CS sniffer rule here prevents any mistakes and issues, so the type would be completely necessary here anyways.

Reply:
For a long time PhpStorm totally sperged out on `self`, it's even only working properly in PHP since PHP 7.4.

Depending on the state of other editors, maybe this can be changed with CakePHP 5, but for 4.x I'd recommend staying with `$this` in order to not break peoples experience now with slightly older editors.

Reply:
True, and Cake4 is still PHP7.2+ (I strongly voted for PHP7.3+ for 4.3 but that still wouldn't change anything^^).

Reply:
Oh, for some reason I actually thought 4.x it was 7.4+, but yeah, with 7.2+ it's out of question as it's not compatible on language level.